runtime/v1/linux/bundle.go

```diff
@@ -127,10 +127,22 @@ func (b *bundle) Delete() error {
 	return errors.Wrapf(err, "Failed to remove both bundle and workdir locations: %v", err2)
 }
 
-func (b *bundle) shimAddress(namespace string) string {
+func (b *bundle) legacyShimAddress(namespace string) string {
 	return filepath.Join(string(filepath.Separator), "containerd-shim", namespace, b.id, "shim.sock")
 }
 
+func (b *bundle) shimAddress() string {
+	return filepath.Join(b.workDir, "shim.sock")
+}
+
+func (b *bundle) decideShimAddress(namespace string) string {
+	shimAddr := b.shimAddress()
+	if _, err := os.Stat(shimAddr); os.IsNotExist(err) {
+		return b.legacyShimAddress(namespace)
+	}
+	return shimAddr
```

Comment:
Does it really work?

Reply:
The new socket path would be $workDir/shim.sock and if it doesn't exist, the client would fallback to use abstract socket. This is for compatible purpose, although it's quite elegant.